spec/unit/puppet/provider/scheduled_task/win32_taskscheduler_spec.rb

```diff
@@ -109,7 +111,20 @@ def time_component
   end
 end
 
-describe Puppet::Type.type(:scheduled_task).provider(:win32_taskscheduler), :if => Puppet.features.microsoft_windows? do
+# The Win32::TaskScheduler and PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2V1Task classes should be
+# API compatible and behave the same way.  What differs is which Windows API is used to query
+# and affect the system.  This means for testing, any tests should be the same no matter what
+# provider or concrete class (which the provider uses) is used.
+klass_list = Puppet.features.microsoft_windows? ? [Win32::TaskScheduler, PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2V1Task] : []
+klass_list.each do |concrete_klass|
+
+if concrete_klass == PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2V1Task
+  task_provider = :taskscheduler_api2
```

Comment:
The one thing I'm not totally keen on is that the constants used in this class are still scoped to `Win32::TaskScheduler::` rather than `concrete_klass::`, so we have a bunch of leakage from v1 provider into v2 provider.

Given what this PR is for now, should be OK for the moment... but we should probably bookmark this as something to loop back on / clean up later. I understand why you want the constants to be the same values for the sake of API compat... 

FWIW, the names of the providers as old `Win32::TaskScheduler` and new `PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2V1Task` don't make it clear that they're related to one another either, but not sure that's something we can or want to change really.

Reply:
I can add a TODO: comment on those

We certainly can't change the 'Win32::TaskScheduler' and that's a hangover of removing the win32-taskscheduler gem.  I strongly feel that changing the name `PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2V1Task` is definitely where it should be, rather than say 'Win32::TaskScheduler2' or something

Reply:
I think what we would want to do is move / rename constants into the new world, and provide aliases from old world to new world for backwards compat, with some timeline on removing the old things (i.e. Puppet 6). It's unlikely that anyone outside  of Pupet is using the old `Win32::TaskScheduler` namespace (we can scan Forge to be absolutely certain though).